torchbenchmark/util/framework/gnn/model_factory.py

```diff
@@ -107,3 +112,40 @@ def eval(self) -> typing.Tuple[torch.Tensor]:
             result = torch.cat(xs, dim=0)
         return (result, )
 
+# Variation of GNNModel based off of test/nn/models/test_basic_gnn.py; the
+# difference is we don't bother with data loading or optimizer step
+class BasicGNNModel(BenchmarkModel):
+    # This benchmark doesn't seem to have any batch size
+    ALLOW_CUSTOMIZE_BSIZE = False
+    DEFAULT_TRAIN_BSIZE = 1
+    DEFAULT_EVAL_BSIZE = 1
+    task = GNN.CLASSIFICATION
+    def __init__(self, model_name, test, device, jit=False, batch_size = None, extra_args=[]):
+        super().__init__(test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+        Model = models_dict[model_name]
+        self.model = Model(64, 64, num_layers=3).to(device)
+        # Apply some global side effects to library (throw out the compiled
+        # model though, we don't need it yet)
+        torch_geometric.compile(self.model)
+        # Make the model jittable
+        # (TODO: This probably makes us overstate the speedup, as making the
+        # model jittable also probably reduces its performance; but this is
+        # matching the benchmark)
+        self.model = sys.modules["torch_geometric.compile"].to_jittable(self.model)
+        num_nodes, num_edges = 10_000, 200_000
+        x = torch.randn(num_nodes, 64, device=device)
+        edge_index = torch.randint(num_nodes, (2, num_edges), device=device)
+        self.example_inputs = (x, edge_index)
+
+    def eval(self):
+        return (self.model(*self.example_inputs),)
+
+    def train(self):
+        # NB: This is a little different than test_basic_gnn.py, as we
+        # are including the cost of randn_like in the overall computation here
+        out = self.model(*self.example_inputs)
+        out_grad = torch.randn_like(out)
```

Comment:
Here our benchmark latency will also include the time we generate the output tensor, but I guess it is fine if the tensor is relatively small.

Optionally we can generate `self.out_grad` in `__init__()` and reference it as `self.out_grad` here, then we won't include tensor generation in the measurement.

Reply:
I think it is more realistic to include the random inside the benchmark